### DISCUSSION TITLE
Could not retrieve access token because: oauth2: cannot fetch token: 401 Unauthorized

### DISCUSSION THREAD

**Discussion by isra270** [ory/hydra#2310] [Q&A] [answered]
I want to connect hydra in golang. 
step 1: set DSN
step 2: set SYSTEM_SECRET
step 3: hydra migrate
step 4: hydra clients create --endpoint http://127.0.0.1:4445/   **to create client** 
**the output :**
OAuth 2.0 Client ID: 4fe65629-a4a2-4851-8cf0-556dbe845666
OAuth 2.0 Client Secret: zlR1VXEN_viVqXz9vlN8QYHFnW
step 5: hydra token client --endpoint http://127.0.0.1:4444/ --client-id 4fe65629-a4a2-4851-8cf0-556dbe845666 --client-secret zlR1VXEN_viVqXz9vlN8QYHFnW
**the output**
Could not retrieve access token because: oauth2: cannot fetch token: 401 Unauthorized
Response: {"error":"invalid_client","error_description":"Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). T
he OAuth 2.0 Client supports client authentication method 'client_secret_basic', but method 'client_secret_post' was requested. You must configure the OAuth 2.0 client's 'token_endpoin
t_auth_method' value to accept 'client_secret_post'."}

**Comment by vinckr** [comment]
Hey isra.
Are you using the binary, docker or Kubernetes?
It says in the response: `
The OAuth 2.0 Client supports client authentication method 'client_secret_basic', but method 'client_secret_post' was requested. You must configure the OAuth 2.0 client's 'token_endpoint_auth_method' value to accept 'client_secret_post'.`

Have a look here: https://www.ory.sh/hydra/docs/cli/hydra-clients-create/#options

**Reply by isra270** [reply]
yes @vinckr  , I am using docker . 
and it works after change this : 
hydra token client --endpoint http://127.0.0.1:4444/ --client-id 4fe65629-a4a2-4851-8cf0-556dbe845666 --client-secret zlR1VXEN_viVqXz9vlN8QYHFnW
to 
hydra token client --endpoint http://127.0.0.1:4444/ --client-id my-client --client-secret secret

**Reply by vinckr** [reply]
But that is not what you want right? 
Does this work?
`hydra token client --endpoint http://127.0.0.1:4444/ --client-id 4fe65629-a4a2-4851-8cf0-556dbe845666 --client-secret zlR1VXEN_viVqXz9vlN8QYHFnW --token-endpoint-auth-method client_secret_post`
Could you give me some more information on your setup? your setup and version, config, log files, docker compose.

**Reply by isra270** [reply]
Thanks so much @vinckr it works.

**Reply by israaaa270** [reply]
> But that is not what you want right?
> Does this work?
> `hydra token client --endpoint http://127.0.0.1:4444/ --client-id 4fe65629-a4a2-4851-8cf0-556dbe845666 --client-secret zlR1VXEN_viVqXz9vlN8QYHFnW --token-endpoint-auth-method client_secret_post`
> Could you give me some more information on your setup? your setup and version, config, log files, docker compose.

`--token-endpoint-auth-method client_secret_post`  this flag is not support for `hydra token` 

I run this but not work.
I need your help.

information :
- I run in terminal hydra binary.
- hydra version:  `oryd/hydra:v1.9.0`
- create client as:  
`hydra clients create --endpoint http://127.0.0.1:4445/ --token-endpoint-auth-method client_secret_post`
Output:
`OAuth 2.0 Client ID: b7e81cae-1013-4b21-a177-2478e2171f95
OAuth 2.0 Client Secret: LeI_7t0E_TPcaqIkk1uOT6B-6d`

--token client as:
`hydra token client --endpoint https://127.0.0.1:4444/ --client-id b7e81cae-1013-4b21-a177-2478e2171f95 --client-secret LeI_7t0E_TPcaqIkk1uOT6B-6d `

the output:

`Could not retrieve access token because: oauth2: cannot fetch token: 401 Unauthorized
Response: {"error":"invalid_client","error_description":"Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). `

**Reply by vinckr** [reply]
Hey @israaaa270 missed your reply here, are you still stuck on this?

**Comment by vinckr** [comment, marked as answer]
Since the topic seems to be stale, I will be selecting an answer for now. 

If you run into the same or a similar issue, feel free to respond in this discussion.

**Reply by israaaa270** [reply]
thank you @vinckr
